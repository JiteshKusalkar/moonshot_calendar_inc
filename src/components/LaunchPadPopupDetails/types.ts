import { Launch } from '../../api/types';

type LaunchPadPopupDetailsProps = {
  launch?: Launch;
};

export type { LaunchPadPopupDetailsProps };
