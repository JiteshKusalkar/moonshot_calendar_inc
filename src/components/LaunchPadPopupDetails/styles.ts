import styled from "styled-components";

const Wrapper = styled.div``;
const ReadonlyField = styled.div`
  display: flex;
  align-self: center;
`;
const Label = styled.span`
  font-weight: 500;
  margin-right: 5px;
`;
const Value = styled.span``;

export { Label, ReadonlyField, Value, Wrapper };
