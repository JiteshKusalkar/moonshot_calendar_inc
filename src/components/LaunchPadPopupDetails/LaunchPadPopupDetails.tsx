import { Label, ReadonlyField, Value, Wrapper } from './styles';
import { LaunchPadPopupDetailsProps } from './types';

function LaunchPadPopupDetails({ launch }: LaunchPadPopupDetailsProps) {
  return (
    <Wrapper>
      <ReadonlyField>
        <Label>Name:</Label>
        <Value>{launch?.pad.location.name || '-'}</Value>
      </ReadonlyField>
      <ReadonlyField>
        <Label>Name of pad:</Label>
        <Value>{launch?.pad.name || '-'}</Value>
      </ReadonlyField>
      <ReadonlyField>
        <Label>Agency:</Label>
        <Value>{launch?.launch_service_provider.name || '-'}</Value>
      </ReadonlyField>
    </Wrapper>
  );
}

export default LaunchPadPopupDetails;
