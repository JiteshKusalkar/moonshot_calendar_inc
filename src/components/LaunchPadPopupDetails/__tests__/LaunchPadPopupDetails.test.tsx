import { render } from '@testing-library/react';
import LaunchPadPopupDetails from '../LaunchPadPopupDetails';

describe('LaunchPadPopupDetails', () => {
  const launch = {
    launch_service_provider: {
      id: 115,
      name: 'Arianespace',
      type: 'Commercial',
    },
    pad: {
      id: 83,
      name: 'Cosmodrome Site 1S',
      latitude: '51.884395',
      longitude: '128.333932',
      location: {
        id: 18,
        url: 'https://lldev.thespacedevs.com/2.2.0/location/18/',
        name: 'Vostochny Cosmodrome, Siberia, Russian Federation',
        country_code: 'RUS',
        map_image:
          'https://spacelaunchnow-prod-east.nyc3.digitaloceanspaces.com/media/launch_images/location_18_20200803142401.jpg',
        total_launch_count: 11,
        total_landing_count: 0,
      },
    },
  };

  it('should render the components with launch details', () => {
    const { container } = render(<LaunchPadPopupDetails launch={launch} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the components without launch details', () => {
    const { container } = render(<LaunchPadPopupDetails />);
    expect(container).toMatchSnapshot();
  });
});
