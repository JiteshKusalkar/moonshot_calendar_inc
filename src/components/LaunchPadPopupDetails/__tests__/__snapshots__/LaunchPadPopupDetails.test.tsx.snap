// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchPadPopupDetails should render the components with launch details 1`] = `
<div>
  <div
    class="sc-bdvvtL"
  >
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Name:
      </span>
      <span
        class="sc-hKwDye"
      >
        Vostochny Cosmodrome, Siberia, Russian Federation
      </span>
    </div>
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Name of pad:
      </span>
      <span
        class="sc-hKwDye"
      >
        Cosmodrome Site 1S
      </span>
    </div>
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Agency:
      </span>
      <span
        class="sc-hKwDye"
      >
        Arianespace
      </span>
    </div>
  </div>
</div>
`;

exports[`LaunchPadPopupDetails should render the components without launch details 1`] = `
<div>
  <div
    class="sc-bdvvtL"
  >
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Name:
      </span>
      <span
        class="sc-hKwDye"
      >
        -
      </span>
    </div>
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Name of pad:
      </span>
      <span
        class="sc-hKwDye"
      >
        -
      </span>
    </div>
    <div
      class="sc-gsDKAQ kCqMgU"
    >
      <span
        class="sc-dkPtRN kTxztz"
      >
        Agency:
      </span>
      <span
        class="sc-hKwDye"
      >
        -
      </span>
    </div>
  </div>
</div>
`;
